sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -424,23 +424,50 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
    */
   protected def stringArgs: Iterator[Any] = productIterator
 
-  /** Returns a string representing the arguments to this node, minus any children */
-  def argString: String = productIterator.flatMap {
-    case tn: TreeNode[_] if containsChild(tn) => Nil
-    case tn: TreeNode[_] => s"${tn.simpleString}" :: Nil
-    case seq: Seq[BaseType] if seq.toSet.subsetOf(children.toSet) => Nil
-    case seq: Seq[_] => seq.mkString("[", ",", "]") :: Nil
-    case set: Set[_] => set.mkString("{", ",", "}") :: Nil
-    case other => other :: Nil
-  }.mkString(", ")
-
-  /** String representation of this node without any children. */
-  def simpleString: String = s"$nodeName $argString".trim
+  /**
+   * Returns a string representing the arguments to this node
+   *
+   * @param verbose Controls whether to output more verbose string or not.
+   */
+  def argString(verbose: Boolean): String = {
+    productIterator.flatMap {
+      case tn: TreeNode[_] if containsChild(tn) => Nil
+      case tn: TreeNode[_] => s"${if (verbose) tn.verboseString else tn.simpleString}" :: Nil
+      case seq: Seq[BaseType] if seq.toSet.subsetOf(children.toSet) => Nil
```

Comment:
I believe `BaseType` here is not checked at compile time and results in a compile time warning.